src/lib/comms/sol-connman-impl-connman.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <systemd/sd-bus.h>
+#include <systemd/sd-event.h>
+
+#define SOL_LOG_DOMAIN &_sol_connman_log_domain
+#include <sol-log-internal.h>
+
+#include "sol-bus.h"
+#include "sol-connman.h"
+
+SOL_LOG_INTERNAL_DECLARE(_sol_connman_log_domain, "connman");
+
+int sol_connman_init(void);
+void sol_connman_shutdown(void);
+
+struct sol_connman_pending {
+    sd_bus_slot *slot;
+};
+
+struct sol_connman_service {
+    struct sol_connman_pending pending;
+    char *path;
+    char *name;
+    enum sol_connman_service_state state;
+    char *type;
+    struct sol_network_link_addr *link_addr;
+    struct sol_network_link_addr *link_addr6;
+    int32_t strength;
+    bool is_call_success;
+};
+
+static void
+_init_connman_service(struct sol_connman_service *service)
+{
+    if (!service)
+        return;
+
+    service->pending.slot = NULL;
+    service->path = NULL;
+    service->type = NULL;
+    service->link_addr = NULL;
+    service->link_addr6 = NULL;
+    service->state = SOL_CONNMAN_SERVICE_STATE_UNKNOWN;
+    service->strength = 0;
+    service->is_call_success = false;
+}
+
+static void
+_free_connman_service(struct sol_connman_service *service)
+{
+    if (!service)
+        return;
+
+    service->pending.slot = sd_bus_slot_unref(service->pending.slot);
+
+    if (service->path)
```

Comment:
You don't need to check `if (x)` before `free(x);`. `free` handles `NULL` input.